File: Array3.java
import java.lang.*;
import java.util.*;

class Array3
{
    public static void main(String arg[])
    {
            int i = 0, j = 0;
            // arr is 2 dimentional array which contains 2 one diemntional array
            // each one dimentional array contains 3 elemnets and each element 
            // is of type integer.
            int arr[][] = new int[2][3];

            arr[0][0] = 11;
            arr[0][1] = 21;
            arr[0][2] = 51;
            arr[1][0] = 101;
            arr[1][1] = 111;
            arr[1][2] = 121;

            // arr.length       2
            // arr[0].langth    3
            // arr[1].length    3
            
            //        1             2              3
            for(i = 0; i < arr.length; i++)     // Outer
            {           //  1               2                3
                    for(j = 0; j < arr[i].length; j++ )     // Inner
                    {
                            System.out.println(arr[i][j]);      // 4
                    }
            }
    }
}

File: Array3.c
#include<stadio.h>

int main()
{
    // Arr is one dimentional array which
    // contains 4 elements and each element
    // is of type integer.
    int Arr[4];

    Arr[0] = 10;
    Arr[1] = 20;
    Arr[2] = 30;
    Arr[3] = 40;

    // Brr is two dimentional array
    // which contains 3 one dimentional array
    // each one dimentional array contains 4 elements
    // and each element is of type integer.
    int Brr[3][4];

    Brr[0][0] = 10;
    Brr[0][1] = 11;
    Brr[0][2] = 12;
    Brr[0][3] = 13;
    Brr[2][1] = 101;


    return 0;
}







// Arr is 2 dimentional array
// which contains 5 one dimentional array
// each one dimentional array contains 3 elements
// each element is of type double.

double Arr[5][3];










int no = 11;

// p is a pointer which holds
// address of integer
// currently it holds address of no

// where no is a variable
// of type integer initialised with the
// value 11

int *p = &no;


//  no                      11
//  &no                     100
//  p                           100
//  &p                         200
//  *p                          11
// sizeof(no)               4
// sizeof(p)                8
// sizeof(*p)               ???












